File: osis/labs/LR-9-10/block-1/main.pas
type
  TIntArray = array[1..10] of Integer; // тип массива целых чисел

// Процедура генерации массивов
// Параметры: a, b — передаются по ссылке (var), изменяются внутри
procedure GenMassive(var a, b: TIntArray);
var
  i: Integer; // локальная переменная
begin
  for i := 1 to 10 do
  begin
    a[i] := Random(101); // случайное число от 0 до 100
    b[i] := Random(101);
  end;
end;

// Процедура вычисления сумм элементов массивов
// Параметры: a, b — по значению (const), sum_a, sum_b — по ссылке (var)
procedure GetSums(const a, b: TIntArray; var sum_a, sum_b: Integer);
var
  i: Integer; // локальная переменная
begin
  sum_a := 0;
  sum_b := 0;

  for i := 1 to 10 do
    sum_a := sum_a + a[i]; // вычисление суммы A

  for i := 1 to 10 do
    sum_b := sum_b + b[i]; // вычисление суммы B
end;

// Процедура проверки и изменения массивов
// Параметры: a, b — по ссылке (var), sum_a, sum_b — по значению
procedure CheckSums(var a, b: TIntArray; sum_a, sum_b: Integer);
var
  i: Integer; // локальная переменная
begin
  if sum_a < sum_b then
  begin
    for i := 1 to 9 do // len(a) - 1
      a[i] := a[i] * 10;
  end
  else
  begin
    for i := 1 to 9 do
      b[i] := b[i] * 10;
  end;
end;

// ГЛАВНАЯ ПРОГРАММА
var
  a, b: TIntArray;     // глобальные переменные (массивы)
  sum_a, sum_b: Integer; // глобальные переменные (суммы)
begin
  GenMassive(a, b);             // фактические параметры
  GetSums(a, b, sum_a, sum_b);  // фактические параметры
  CheckSums(a, b, sum_a, sum_b);// фактические параметры

  // вывод результатов
  writeln('Массив A: ', a);
  writeln('Массив B: ', b);
  Writeln('Сумма A: ', sum_a);
  Writeln('Сумма B: ', sum_b);
end.

File: osis/labs/LR-9-10/block-1/main.py
import random  # глобальный импорт модуля

def gen_massive():
    # локальные переменные функции
    a = [random.randint(0, 100) for _ in range(10)]  # список случайных чисел от 0 до 100
    b = [random.randint(0, 100) for _ in range(10)]
    return a, b  # возвращаем два списка (ссылкой)


def get_sums(a: list, b: list):
    # формальные параметры: a, b (передаются по ссылке, т.к. это списки)
    # локальные переменные функции:
    sum_a, sum_b = 0, 0

    # цикл по элементам списка a
    for i in a:  # i — локальная переменная цикла
        sum_a += i

    # цикл по элементам списка b
    for i in b:
        sum_b += i

    return sum_a, sum_b  # возвращаем результаты (по значению, т.к. это числа)


def check_sums(a: list, b: list, sum_a: int, sum_b: int):
    # формальные параметры: a, b, sum_a, sum_b
    # a, b — передаются по ссылке (могут изменяться внутри функции)
    # sum_a, sum_b — передаются по Ссылке (числа, неизменяемые типы)

    if sum_a < sum_b:
        for i in range(len(a) - 1):  # i — локальная переменная цикла
            a[i] = a[i] * 10 
    else:
        for i in range(len(b) - 1):
            b[i] = b[i] * 10


# глобальные переменные a и b — создаются в основной программе
a, b = gen_massive()           # Глобальнык переменные массивов
sum_a, sum_b = get_sums(a, b)  # глобальные переменные для хранения сумм
check_sums(a, b, sum_a, sum_b)

# вывод результатов
print("Массив A:", a)
print("Массив B:", b)
print("Сумма A:", sum_a)
print("Сумма B:", sum_b)
